src/Aquifer.API/Endpoints/Bibles/Versification/Endpoint.cs

```diff
@@ -0,0 +1,192 @@
+        {
+            return (req.StartChapter!.Value, maxChapter);
+        }
+        
+        var minChapter = await dbContext.BookChapters
+            .Where(bc => bc.BookId == (BookId)req.BookId)
+            .MinAsync(bc => bc.Number, ct);
+
+        if (!isRequestedStartChapterValid && isRequestedEndChapterValid)
+        {
+            return (minChapter, req.EndChapter!.Value);
+        }
+        
+        return (minChapter, maxChapter);
+    }
+    
+    private async Task<(int, int)> GetMinAndMaxVerseIdFromRequestAsync(Request req, CancellationToken ct)
+    {
+        var (minChapterNumber, maxChapterNumber) = await GetMinAndMaxChapterNumberFromRequestAsync(req, ct);
+        
+        var isRequestedStartVerseValid = false;
+        var isRequestedEndVerseValid = false;
+
+        if (req.StartVerse.HasValue)
+        {
+            isRequestedStartVerseValid = await IsVerseValidForBookAndChapterAsync(req.StartVerse.Value, minChapterNumber, req.BookId, ct);
+        }
+
+        if (req.EndVerse.HasValue)
+        {
+            isRequestedEndVerseValid = await IsVerseValidForBookAndChapterAsync(req.EndVerse.Value, minChapterNumber, req.BookId, ct);
+        }
+        
+        if (isRequestedStartVerseValid && isRequestedEndVerseValid)
+        {
+            return (BibleUtilities.GetVerseId((BookId)req.BookId, minChapterNumber, req.StartVerse!.Value),
+                    BibleUtilities.GetVerseId((BookId)req.BookId, maxChapterNumber, req.EndVerse!.Value));
+        }
+        
+        var maxVerse = await dbContext.BookChapters
+            .Where(bc => bc.BookId == (BookId)req.BookId && bc.Number == maxChapterNumber)
+            .Select(bc => bc.MaxVerseNumber)
+            .FirstOrDefaultAsync(ct);
+        
+        if (isRequestedStartVerseValid && !isRequestedEndVerseValid)
+        {
+            return (BibleUtilities.GetVerseId((BookId)req.BookId, minChapterNumber, req.StartVerse!.Value),
+                    BibleUtilities.GetVerseId((BookId)req.BookId, maxChapterNumber, maxVerse));
+        }
+        
+        var minVerse = await dbContext.BibleTexts
+            .Where(bt => bt.BookId == (BookId)req.BookId && bt.ChapterNumber == minChapterNumber)
+            .OrderBy(bt => bt.VerseNumber)
+            .Select(bt => bt.VerseNumber)
+            .FirstOrDefaultAsync(ct);
+
+        if (!isRequestedStartVerseValid && isRequestedEndVerseValid)
+        {
+            return (BibleUtilities.GetVerseId((BookId)req.BookId, minChapterNumber, minVerse),
+                    BibleUtilities.GetVerseId((BookId)req.BookId, maxChapterNumber, req.EndVerse!.Value));
+        }
+        
+        return (BibleUtilities.GetVerseId((BookId)req.BookId, minChapterNumber, minVerse),
+                BibleUtilities.GetVerseId((BookId)req.BookId, maxChapterNumber, maxVerse));
+    }
+    
+    private static VerseReference FormatBaseBookChapterVerseMapping(int mappedVerseId)
```

Comment:
Nit: I'd probably name this with a `Map` verb.  If you disagree that's fine.
```suggestion
    private static VerseReference MapToVerseReference(int verseId)
```

Reply:
This is good. that format name was a holdover from the previous iteration when returning a string